src/http_rest_api_handler.cpp

```diff
@@ -163,7 +163,7 @@ void HttpRestApiHandler::registerHandler(RequestType type, HandlerCallbackFn f)
 }
 
 void HttpRestApiHandler::registerAll() {
-    registerHandler(Predict, [this](const std::string_view uri, const HttpRequestComponents& request_components, std::string& response, const std::string& request_body, HttpResponseComponents& response_components, std::shared_ptr<HttpAsyncWriter> serverReaderWriter) -> Status {
+    registerHandler(Predict, [this](const std::string_view uri, const HttpRequestComponents& request_components, std::string& response, const std::string& request_body, HttpResponseComponents& response_components, std::shared_ptr<HttpAsyncWriter> serverReaderWriter, std::shared_ptr<MultiPartParser> multiPartParser) -> Status {
```

Comment:
Why this need to be shared_ptr?

Reply:
this is because we may launch graph executor asynchronously when streaming
as of now the parser is not used after asynchronous work is started, however we should not limit the caller to make use of this parser in future, for reference:

https://github.com/openvinotoolkit/model_server/blob/multipart-rb/src/http_server.cpp#L210-L262

@atobiszei